ext/CreateParametersExt.jl

```diff
@@ -0,0 +1,137 @@
+module CreateParametersExt
```

Comment:
Thanks @nefrathenrici, are there any limitations to this that we should be aware of? Perhaps we add a link to `https://pkgdocs.julialang.org/v1.9/creating-packages/#Conditional-loading-of-code-in-packages-(Extensions)` to help users.

Reply:
This is fully backwards-compatible for Julia <1.9.
For 1.9+, CLIMAParameters needs to be loaded beforehand to compile the constructors in the extension. Without this step, users won't have access to the constructors. 
I'm happy to add the link and an explanation for creating parameters to the docs.

Reply:
Noted, thanks @nefrathenrici